threadit-web/src/api/SpoolAPI.ts

```diff
@@ -70,4 +77,92 @@ export default class SpoolAPI {
 
         return await response.json();
     }
+
+    static async getAllMods(spoolId: string): Promise<IUserProfile[]> {
+        const response = await get(allModsForSpoolEndpoint + spoolId);
+
+        if (!response.ok) {
+            throw new Error(`Failed to get all mods for the spool: ${await response.text()}`);
+        }
+
+        return await response.json();
+    }
+
+    static async removeModerator(spoolId: string, userId: string): Promise<IUserProfile> {
+        const response = await get(removeModeratorEndpoint + spoolId + '/' + userId);
+
+        if (!response.ok) {
+            throw new Error(`Failed to remove moderator from spool: ${await response.text()}`);
+        }
+
+        return await response.json();
+    }
+
+    static async addModerator(spoolId: string, userName: string): Promise<number> {
+        const response = await get(addModeratorEndpoint + spoolId + '/' + userName);
+        if (response.ok) {
+            //returns 1 if the add was successful
+            return 1;
+        }
+
+        else {
+            //otherwise change failed and need to see why
+            const errorText = await response.text();
+            if (errorText === "User does not exist.") {
```

Comment:
why not just return the text? and `true` if it succeeded.

return type is then `Promise<bool | string>`

that way you dont need your if statements when you display the error later

Reply:
because I didnt know you could do that lol